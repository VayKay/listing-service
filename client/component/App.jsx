import React from 'react';
const faker = require('faker');
import $ from 'jquery';
import Description from './Description';


var num = faker.random.number({min:1, max:100});
class App extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            id: num,
            title: '',
            location: '',
            host: {},
            highlights: {},
            desc: {},
            detail: {},
            amenity: {},
        }
    }
    componentDidMount(){
        
        $.ajax({
            method:'GET',
            url: '/listing/desc/' + this.state.id,
            contentType: 'application/json',
            success: (desc)=>{
                // console.log(desc)
                this.setState({
                    title: desc.title,
                    location: desc.location,
                    host: desc.host,
                    highlights: desc.highlights,
                    desc: desc.desc,
                    detail: desc.detail    
                })
                
            }
        })
        $.ajax({
            method:'GET',
            url: '/listing/amenity/' + this.state.id,
            contentType: 'application/json',
            success: (amenity)=>{
                // console.log('amenity',amenity)
                this.setState({
                    amenity: amenity.amenities
                })
                
            }
        })
    }
    render() {
        // console.log('id',this.state)
        return (
            <div id='main'>
                <Description  title={this.state.title} location={this.state.location} host={this.state.host} highlights={this.state.highlights} desc={this.state.desc} detail={this.state.detail} amenity={this.state.amenity}/>
            </div>
        )
    }
}
    
export default App;