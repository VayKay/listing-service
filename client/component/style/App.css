.main {
    margin-left: auto;
    margin-right: auto;
    position: relative;
    max-width: 696px;
    padding-right: 24px;
    padding-left: 24px;
}