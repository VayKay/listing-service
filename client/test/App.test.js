import React from 'react';
import App from '../component/App.jsx'
import { shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
require('babel-core/register');
require('babel-polyfill');

configure({ adapter: new Adapter() });

describe('App test suite', () => {
    it('render App' , ()=>{
        const app = shallow(<App />);
        expect(app.exists()).toBe(true);
    })
    it('snapshots', async()=>{
        const app = shallow(<App />);
        expect(toJson(app)).toMatchSnapshot()
    })
    it('render child div' , ()=>{
        const app = shallow(<App />);
        expect(app.exists('#main')).toBeTruthy();
    })
});