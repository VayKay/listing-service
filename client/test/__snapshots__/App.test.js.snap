// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test suite snapshots 1`] = `
<div
  id="main"
>
  <Description
    amenity={Object {}}
    desc={Object {}}
    detail={Object {}}
    highlights={Object {}}
    host={Object {}}
    location=""
    title=""
  />
</div>
`;
